crypto/bloom/timestamp_bloom_test.go

```diff
@@ -0,0 +1,122 @@
+package bloom_test
+
+import (
+	"bytes"
+	"github.com/orbs-network/orbs-network-go/crypto/bloom"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"testing"
+	"time"
+)
+
+type tsbfNewTrio struct {
+	input    int
+	size     uint32
+	bitCount uint8
+}
+
+var newTestTable = []tsbfNewTrio{
+	{input: 15, size: 16, bitCount: 4},
+	{input: 20, size: 32, bitCount: 5},
+	{input: 200, size: 256, bitCount: 8},
+	{input: 1024, size: 1024, bitCount: 10},
+}
+
+var nanoForRaw = []primitives.TimestampNano{
+	1533731643509419667,
+	1533731643509435135,
+	1533731643509465891,
+	1533731643509489416,
+	1533731643509515475,
+	1533731643509519636,
+	1533731643511038190,
+	1533731643511049004,
+}
+
+func TestNew(t *testing.T) {
+	for _, test := range newTestTable {
+		x := bloom.New(test.input)
+		if x.Size() != test.size {
+			t.Errorf("size should be %d, but is %d", test.size, x.Size())
+		}
+
+		if x.BitCount() != test.bitCount {
+			t.Errorf("bitcount should be %d, but is %d", test.bitCount, x.BitCount())
+		}
+	}
+}
+
+func TestTimestampBloomFilter_Add(t *testing.T) {
```

Comment:
Bloom filters might be a good opportunity to play with Property Based Testing, which essentially means running your invariants against a large set of randomly-generated test data.

A quick Google finds this:
https://godoc.org/github.com/leanovate/gopter

Maybe have a look?

Reply:
good idea, the current implementation is not a real one, its more of a stub then something we should vigorously test - we should add this as a requirement for the real implementation (pending some research talks, will hopefully be implementing by the end of August)